utils/unittest/entity_test.go

```diff
@@ -0,0 +1,162 @@
+package unittest
+
+import (
+	"testing"
+
+	"github.com/onflow/crypto"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/slices"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type IntList []uint32
+
+func (l IntList) ID() flow.Identifier {
+	return flow.MakeID(l)
+}
+
+type StructWithNilFields struct {
+	Identities flow.IdentitySkeletonList
+	Index      map[flow.Identifier]uint32
+	QC         *flow.QuorumCertificate
+}
+
+func (e *StructWithNilFields) ID() flow.Identifier {
+	type pair struct {
+		ID    flow.Identifier
+		Index uint32
+	}
+	var pairs []pair
+	for id, index := range e.Index {
+		pairs = append(pairs, pair{id, index})
+	}
+	slices.SortFunc(pairs, func(a, b pair) int {
+		return flow.IdentifierCanonical(a.ID, b.ID)
+	})
+	return flow.MakeID(struct {
+		Identities flow.IdentitySkeletonList
+		Index      []pair
+		QcID       flow.Identifier
+	}{
+		Identities: e.Identities,
+		Index:      pairs,
+		QcID:       e.QC.ID(),
+	})
+}
+
+type StructWithUnsupportedFlowField struct {
```

Comment:
```suggestion
// StructWithUnsupportedFlowField will always fail malleability checking because it contains a private (non-settable) field
type StructWithUnsupportedFlowField struct {
```